lms/views/api/canvas/files.py

```diff
@@ -43,13 +45,22 @@ def __init__(self, request):
     @view_config(request_method="GET", route_name="canvas_api.courses.files.list")
     def list_files(self):
         """Return the list of files in the given course."""
-        return self.canvas_api_client.list_files(self.request.matchdict["course_id"])
+        try:
+            return self.canvas_api_client.list_files(
+                self.request.matchdict["course_id"]
+            )
+        except CanvasAPIError as err:
+            raise HTTPInternalServerError(explanation=err.explanation) from err
```

Comment:
I had a query about whether this would produce a different result than not catching the exception at all. @seanh explained [on Slack](https://hypothes-is.slack.com/archives/C4K6M7P5E/p1561036091010000) that the difference is that the `explanation` will be presented to the user, whereas exception info for an uncaught exception is translated into a generic message for the user, in case it contains information that shouldn't be exposed.